dashboard/app/models/teacher_based_experiment.rb

```diff
@@ -29,14 +29,25 @@ class TeacherBasedExperiment < Experiment
   validates :percentage, inclusion: 0..100
 
   def enabled?(user: nil, section: nil)
-    return false unless section
+    is_teacher = user.try(:teacher?)
+    return false unless section || is_teacher
+
+    if is_teacher
+      user_id = user.id
+      sections = user.sections
+    else
+      user_id = section.user_id
+      sections = [section]
+    end
```

Comment:
Though I'm not certain of what logic is desired, this doesn't work as I'd expect. My expectation is that the experiment would be enabled if the user is an section (as a student) for which the experiment is enabled OR if the user is in a section (as a teacher) for which the experiment is enabled. This seems to only check one.

And I don't understand why we need to gate the logic on `is_teacher`, doing `user.sections` is fine for students (it should be empty).